### PR TITLE
Fix "/Schemas" endpoint; configurable default resources

### PR DESCRIPTION
**Pull request by pond**
* Fixes https://github.com/RIPAGlobal/scimitar/issues/117
* Implements https://github.com/RIPAGlobal/scimitar/issues/118
* Drive-by `README.md` updates that were overlooked after some prior enhancements
* Drive-by update to use standard Ruby debugger instead of `byebug` during development

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -632,20 +649,6 @@ If you believe choices made in this section may be incorrect, please [create a G
...
-
-  Another filter shows a potential workaround:
-
-  - `filter=userType eq "Employee" and (emails.type eq "work")`
-
-  ...which is just a match on `emails.type`, so if you have a queryable attribute mapping defined for `emails.type`, that would become queryable. Likewise, you could rewrite the more complex prior example thus:
-
-  - `filter=userType eq "Employee" and emails.type eq "work" and emails.value co "@example.com"`
-
-  ...so adding a mapping for `emails.value` would then allow a database query to be constructed.
```
Comment (pond) [inline README.md:647]: The topmost example does now work (and I've added test coverage to prove it). Unsure exactly when this started working. It looks a bit like https://github.com/RIPAGlobal/scimitar/issues/115, but isn't.

File: lib/scimitar/engine.rb [hunk 2]
```diff
@@ -47,23 +63,44 @@ def self.reset_custom_resources
...
-      @custom_resources ||= []
-    end
-
-    # Returns the default resources added in this gem:
+    # Returns the default resources added in this gem - by default, these are:
     #
     # * Scimitar::Resources::User
     # * Scimitar::Resources::Group
     #
+    # ...but if an implementation does not e.g. support Group, it can override
```
Comment (bagp1) [inline lib/scimitar/engine.rb:71]:
```suggestion
    # ...but if an implementation does not e.g. support Group, it can overriden
```

**Review by bagp1 — APPROVED** [review]
Nice work. I found a single incredibly minor grammatical mistake in a comment, but this is otherwise good to go.

**Comment by gsar** [comment]
great work here @pond, thanks for taking it on!

File: README.md [hunk 3]
```diff
@@ -582,6 +582,23 @@ And write to it like this:
...
+
+If you have an API consumer entity querying your Scimitar-based SCIM API provider endpoint and want to enable a degree of auto-discovery for that entity, then depending on your implementation, there may be customisations you wish to make.
+
+#### Default resources
+
+By default, Scimitar advertises (via things like [the `/Schemas` endpoint](https://tools.ietf.org/html/rfc7644#section-4)) support for both a `User` and `Group` resource, but if you (say) only support a `User` concept, you override the default using code such as this in your `config/initializers/scimitar.rb` file:
+
+```ruby
+Rails.application.config.to_prepare do
+  Scimitar::Engine::set_default_resources([Scimitar::Resources::User])
```
Comment (gsar) [inline README.md:595]: @pond although this appears to work, there are linter complaints about writing it the more idiomatic way:

``` ruby
  Scimitar::Engine.set_default_resources([Scimitar::Resources::User])
```

Comment (pond) [reply README.md:595]: Yeah, fair. Thanks. I'll do a PR in the near-ish future that's just a style change for the above onto `main`/`v1` but won't release a new gem version given there's no functional change.

**Pull request merged**